### ISSUE TITLE
False positive on new register projections

### ISSUE THREAD

**Issue by sibebleuze** [ansible/ansible-lint#5067]
##### Summary

Ansible-lint throws a type error on the register keyword when the new register projections feature (ansible-core 2.21) is used.
I believe this is due to the linting schema (see extract below of the schema on main at time of issue creation) which has the type set to string, whereas the new register projections feature allows it to be an object (like `vars`).

https://github.com/ansible/ansible-lint/blob/21cbf710ef70a4c16d3695223a0143261c3e024a/src/ansiblelint/schemas/ansible.json#L1100-L1102

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 26.4.0 using ansible-core:2.21.0 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:None
```

- ansible installation method: homebrew on MacOS
- ansible-lint installation method: homebrew on MacOS

##### STEPS TO REPRODUCE

- Use the new register projections feature (examples [here](https://github.com/ansible/ansible/pull/86241) or below).
- Run ansible-lint on the file where this is used.

```console (paste below)
- name: Ask for username
  ansible.builtin.pause:
    prompt: "Enter username [default: my_username]"
  register:
    username: _task.result.user_input | ansible.builtin.default('my_username', true)
```

##### Desired Behavior

Ansible-lint does not see any issues with the task

##### Actual Behavior

VSCode hint when hovering over the task with the issue
```
Incorrect type. Expected "string". yaml-schema: Register
```

Output of `ansible-lint main.yml`
```paste below
schema[tasks]: $[1].register {'username': "_task.result.user_input | ansible.builtin.default('my_username', true)"} is not of type 'string'[/]
```

**Comment by sibebleuze** [comment]
This is extra annoying because the `schema` rule cannot be ignored (with `noqa`), so the only 2 options I have are
- disable linting completely
- revert to the old `register` + `set_fact` behaviour

**Comment by sibebleuze** [comment]
@ssbarnea could you point me to the PR or ansible-lint version this got fixed in?
